### ISSUE TITLE
Some sequences silently don't work

### ISSUE THREAD

**Issue by gitoliver** [GLYCAM-Web/GMML#148]
DSorp[3S]a2-4DGalpA[3D]b1-2LIdopAa1-3DXylpa1-2DManp[4A,3A]a1-3DGalpb1-4[DArapa1-2]LFucpa1-3DGalpNAca1-OH
I get Project Status : Finished and all buttons become clickable. But the downloaded Download all file contains only info.json, structure.pdb and structure.off. It's a single rotamer, the info.json looks like this:
  "payload": "",
  "pUUID": "4f1256c0-dcc0-4771-8308-2f944a5e7836",
  "entity_id": "sequence",
  "service_id": "cb",
  "incomingSequence": "",
  "indexOrderedSequence": "DArapa1-2[D3[DSorp[3S]a2-4]DGalpAb1-2LIdopAa1-3DXylpa1-2DManp[3A,4A]a1-3DGalpb1-4]LFucpa1-3DGalpNAca1-OH",
  "seqID": "",
  "conformerID": "structure",
  "conformerLabel": "structure",
  "sequenceConformation": [],
  "isDefaultStructure": true,
  "isNewBuild": true,
  "structureDirectoryName": "structure",
  "filesystem_path": "/website/userdata/",
  "host_url_base_path": "https://test.glycam.org",
  "conformer_path": "/website/userdata/sequence/cb/Builds/4f1256c0-dcc0-4771-8308-2f944a5e7836/Requested_Builds/structure",
  "absolute_conformer_path": "/website/userdata/sequence/cb/Builds/4f1256c0-dcc0-4771-8308-2f944a5e7836/New_Builds/structure",
  "downloadUrlPath": "https://test.glycam.org/json/download/sequence/cb/4f1256c0-dcc0-4771-8308-2f944a5e7836/structure/",
  "forceField": "See Build Directory Files"
}
~    

The pdb file looks like this:
TITLE     Generated by GMML
END
~

**Comment by gitoliver** [comment]
This works on my development branch, which will be merged into gmml-test soon. Probably best to wait for that and then close.

**Comment by gitoliver** [comment]
Still getting stuck on the download page of actual and test with minimiztion as the status. Need to dig a bit.

**Comment by gitoliver** [comment]
It was working in gmmls carbohydrate builder test as it uses the string that you pass in. At the gems level it first orders the sequence to find sequences that result in the same 3D structure. Note the D3 (deoxy on the 3 position) moving to the wrong place in the index ordered version:
DArapa1-2[**D3**[DSorp[3S]a2-4]DGalpAb1-2LIdopAa1-3DXylpa1-2DManp[3A,4A]a1-3DGalpb1-4]LFucpa1-3DGalpNAca1-OH
When this is passed into gmml, it throws an error. 
I've found the reason for this in gmml and will push a fix to test. This uncovered a different bug in gems/website though, the error from gmml was not propagated correctly.

**Comment by gitoliver** [comment]
The sequence was working in gmmls carbohydrate builder test as it uses the string that you pass in. At the gems level it first orders the sequence to find sequences that result in the same 3D structure. Note the D3 (deoxy on the 3 position) moving to the wrong place in the index ordered version:
DArapa1-2[**D3**[DSorp[3S]a2-4]DGalpAb1-2LIdopAa1-3DXylpa1-2DManp[3A,4A]a1-3DGalpb1-4]LFucpa1-3DGalpNAca1-OH
When this is passed into gmml, it throws the error we see in the gems logs. 
I've found the reason for this in gmml and pushed a fix to gmml-test. This uncovered a different bug in gems/website though, the error from gmml was not propagated correctly. I've opened another bug ticket in gems.
